runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricUpdates.java

```diff
@@ -66,21 +68,30 @@ public static <T> MetricUpdate<T> create(MetricKey key, T update) {
   /** All the sets updates. */
   public abstract Iterable<MetricUpdate<StringSetData>> stringSetUpdates();
 
+  /** All the histogram updates. */
+  public abstract Iterable<MetricUpdate<HistogramData>> perWorkerHistogramsUpdates();
+
   /** Create a new {@link MetricUpdates} bundle. */
   public static MetricUpdates create(
       Iterable<MetricUpdate<Long>> counterUpdates,
       Iterable<MetricUpdate<DistributionData>> distributionUpdates,
       Iterable<MetricUpdate<GaugeData>> gaugeUpdates,
-      Iterable<MetricUpdate<StringSetData>> stringSetUpdates) {
+      Iterable<MetricUpdate<StringSetData>> stringSetUpdates,
+      Iterable<MetricUpdate<HistogramData>> perWorkerHistogramsUpdates) {
```

Comment:
Other metrics aren't "perWorker" but this one does which sounds strange to me.

I would assume this MetricUpdates.create is created on bundle basis, corresponds to each MetricsContainer. If we insert a per-worker metrics here would it be reported multiple times by different bundles for the same worker?

Reply:
and in general we may need to think more about perWorker metrics in portable runner. This PR involves Beam model change, I would suggest open a GitHub issue and/or involve people cc: @robertwb

Reply:
talked offline. The original comment was about the change added a metrics named "per worker" but still based on per bundle metrics infra, now I understand the per-worker accumulation is going to happen in runner (Dataflow runner v2) side. If this is the case, we should name it as "histogramsUpdates" in align with other existing metrics in Beam spec, because "per worker" accumulation is a runner implementation, not controlled in SDK side (not part of Beam spec).